api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java

```diff
@@ -155,8 +159,13 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
     }
     DbUser user = userDao.findUserByUsername(userName);
     if (user == null) {
-      // TODO(danrodney): start populating contact email in Google account, use it here.
-      user = userService.createUser(OAuth2Userinfo);
+      if (workbenchConfigProvider.get().access.unsafeAllowUserCreationFromGSuiteData) {
+        DbUser dbUser = devUserRegistrationService.createUserFromUserInfo(userInfo);
+        log.info(String.format("Dev user '%s' has been re-created.", dbUser.getUsername()));
```

Comment:
for later: it would be good to audit this...though if it never happens in production 🤷 .